python-avd/pyavd/_schema/models/avd_indexed_list.py

```diff
@@ -297,3 +299,31 @@ def _compare(self, other: object) -> bool:
             return False
 
         return all(item._compare(other[key]) for key, item in self.items())
+
+    def _combine(self, other: Self) -> None:
+        """
+        Update instance by combining the other instance in.
+
+        Combining is different from merging in the sense that it will raise if there is a conflict
+        between one of our elements and the other elements.
+
+        for AvdIndexedList this is simply an append, if any conflict occurs a duplicate error will be raised.
+
+        Args:
+            other: The other instance of the same type to combine into this instance.
+
+        Raises:
+            AristaAvdDuplicateDataError: If any item from other is conflicting with an item from self when appending.
+        """
+        cls = type(self)
+        if not isinstance(other, cls):
+            msg = f"Unable to combine type '{type(other)}' into '{cls}'"
+            raise TypeError(msg)
+
+        # If the value of _created_from_null are different, consider it as a conflict.
+        if self._created_from_null != other._created_from_null:
+            raise AristaAvdDuplicateDataError(type(self).__name__, str(self._dump()), str(other._dump()))
+
+        for item in other:
+            # TODO: Do we need to copy the item to prevent some unsavory things later if other is modified?
```

Comment:
I don't think we do that for merging, so we should not here either. It was a general performance optimization.